src/springfield/fields.py

```diff
@@ -390,12 +390,26 @@ def __init__(self, entity, *args, **kwargs):
                        Use 'self' to use the :class:`Entity` class that
                        this field is already bound to.
         """
-        self.type = entity
+        self._type = entity
         super(EntityField, self).__init__(*args, **kwargs)
 
+    @property
+    def type(self):
+        if not callable(self._type):
+            try:
+                if '.' in self._type:
+                    modules, _kls_name = self._type.rsplit('.', 1)
+                    _module = __import__(modules, fromlist=[_kls_name])
+                    _kls = getattr(_module, _kls_name) 
+                    assert callable(_kls), 'Dotted-name entity types should be callable.'
+                    self._type = _kls
+            except Exception as e:
+                raise ValueError('Invalid class name for EntityField: %s (%s)' % (self._type, e))
+        return self._type
+
     def init(self, cls):
-        if self.type == 'self':
-            self.type = cls
+        if self._type == 'self':
+            self._type = cls
```

Comment:
Couldn't the string/import happen here? I can't remember where in the process `init` is called.